.travis.yml

```diff
@@ -6,45 +6,62 @@ cache:
   directories:
     - $HOME/.composer/cache
 
+php:
+  - 7.0
+  - 7.1.18
+  - 7.2
+  - nightly
+  - hhvm
+
+env:
+  - SYMFONY_VERSION=2.7.*
+  - SYMFONY_VERSION=2.8.*
+  - SYMFONY_VERSION=3.0.*
+  - SYMFONY_VERSION=3.1.*
+  - SYMFONY_VERSION=3.2.*
+  - SYMFONY_VERSION=3.3.*
+  - SYMFONY_VERSION=3.4.*
+  - SYMFONY_VERSION=4.0.*
+  - SYMFONY_VERSION=4.1.*
+  - DEPENDENCIES="beta"
+
 matrix:
-  include:
+  fast_finish: true
+  exclude:
     - php: 7.0
-      env: SYMFONY_VERSION=3.2.*
+      env: SYMFONY_VERSION=4.0.*
     - php: 7.0
-      env: SYMFONY_VERSION=3.3.*
+      env: SYMFONY_VERSION=4.1.*
     - php: 7.0
-      env: DEPENDENCIES=beta
-    - php: 7.1
+      env: DEPENDENCIES="beta"
+    - php: 7.1.18
       env: SYMFONY_VERSION=3.2.*
-    - php: 7.1
+    - php: 7.1.18
       env: SYMFONY_VERSION=3.3.*
-    - php: 7.1
-      env: DEPENDENCIES=beta
     - php: nightly
       env: SYMFONY_VERSION=3.2.*
     - php: nightly
       env: SYMFONY_VERSION=3.3.*
-    - php: nightly
-      env: DEPENDENCIES=beta
   allow_failures:
     - php: nightly
-
-env:
-  global:
-    - deps=no
+    - php: hhvm
+    - env: DEPENDENCIES="beta"
 
 before_install:
-  - composer self-update
-  # Set composer minimum-stability configuration filter to beta versions
-  - if [ "$DEPENDENCIES" = "beta" ]; then perl -pi -e 's/^}$/,"minimum-stability":"beta"}/' composer.json; fi;
+  - composer install --no-interaction
```

Comment:
@martin-georgiev this should move to the end of the install definitions, the `composer config minimum-stability` and `composer require` at present won't work (separate comment regarding `require`).

Reply:
@martin-georgiev this should probably also be `composer update -n --prefer-dist` again as no lock file